src/environment.js

```diff
@@ -99,7 +97,14 @@ var Environment = Obj.extend({
         return this.filters[name];
     },
 
+    resolveTemplate: function(loader, parentName, filename) {
+        var isRelative = (loader.isRelative && parentName)? loader.isRelative(filename) : false;
+        return (isRelative && loader.resolve)? loader.resolve(parentName, filename) : filename;
```

Comment:
This is related to #381, let me know if I should ignore the issue or keep the exist check.